test/functional/tests/misc/test_random_cache_ops.py

```diff
@@ -0,0 +1,632 @@
+from api.cas.cli_messages import (
+    start_cache_with_existing_metadata,
+    check_stderr_msg,
+    no_cas_metadata,
+    start_cache_with_existing_id,
+    attach_cache_with_existing_metadata,
+)
+from connection.utils.output import CmdException
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite
+from datetime import datetime, timedelta
+
+from type_def.size import Size, Unit
+from type_def.time import Time
+
+io_engine_list = [
+    IoEngine.sync,
+    IoEngine.libaio,
+    IoEngine.psync,
+    IoEngine.pvsync,
+    IoEngine.posixaio,
+    IoEngine.mmap,
+]
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_random_operations():
+    """
+    title: Stress test for allowed operations.
+    description: |
+        Test running all allowed operation multiple times in random order.
+    pass_criteria:
+      - No system crash.
+    """
+    test_runtime = timedelta(minutes=60)
+
+    with TestRun.step("Prepare disks"):
+        cache_dev = TestRun.disks["cache"]
+        core_dev = TestRun.disks["core"]
+
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)] * 5)
+        core_dev.create_partitions([Size(2, Unit.GibiByte)] * 10)
+
+    with TestRun.step("Start cache and add core"):
+        main_cache = casadm.start_cache(cache_dev=cache_dev.partitions[0], force=True)
+        main_cache.add_core(core_dev=core_dev.partitions[0])
+
+    with TestRun.step("Run random operations"):
+        used_cache_devices = [cache_dev.partitions[0]]
+        available_cache_devices = cache_dev.partitions[1:]
+        used_core_devices = [core_dev.partitions[0]]
+        available_core_devices = core_dev.partitions[1:]
+        operation_list = [
+            start_cache,
+            attach_cache,
+            detach_cache,
+            stop_cache,
+            set_param,
+            set_cache_mode,
+            add_core,
+            remove_core,
+            remove_inactive,
+            reset_counters,
+            flush_cache,
+            zero_metadata,
+            run_fio,
+        ]
+
+        # set runtime of test
+        test_start_time = datetime.now()
+        step_name_list = []
+        counter = 0
+
+        try:
+            while datetime.now() < test_start_time + test_runtime:
+                counter += 1
+                step = random.choice(operation_list)
+                step_name = step(
+                    available_cache_devices,
+                    used_cache_devices,
+                    available_core_devices,
+                    used_core_devices,
+                )
+                if step_name:
+                    step_name_list.append(step.__name__)
+        except CmdException:
+            step_name_list.append(step.__name__)
```

Comment:
I meant the step.\_\_name\_\_ only for the exception handling - that is the case when step() throws an Exception and doesn't return anything (so step_name is still set by the previous step)
```suggestion
                if step_name:
                    step_name_list.append(step_name)
        except CmdException:
            step_name_list.append(step.__name__)
```